localstack/services/events/provider.py

```diff
@@ -596,12 +688,21 @@ def _get_limited_dict_and_next_token(
         return limited_dict, next_token
 
     def _extract_event_bus_name(
```

Comment:
This method is getting increasingly complex. Could be a candidate for a unit test to cover edge cases easily (as long as we can ensure AWS parity).